docs/adapter_impl_notes.md

```diff
@@ -0,0 +1,184 @@
+# Adapter Implementation Notes
+
+This page talks about implementation details of writing adapters. If you want to write your own adapter for your own backend, use this document as a guide.
+
+For this document, we'll use the Gmail Adapter as an example. In this case, the backend consists a set of email messages. Each message contains a timestamp (when the message was received) and fields (email headers such as ``From:``, ``To:``, ``Subject:``, the message body, etc). The Gmail API supports [search expressions](https://support.google.com/mail/answer/7190?hl=en) that select messages based on date, headers, or a free-text search on the message contents.
+
+As of 2015-12-23, the adapter implementation doesn't *exactly* match this description. For example, the adapter doesn't yet support filtering expressions or live programs. However, the notes still do a good job of illustrating the concepts.
+
+The adapter's job is to interpret the options included in the juttle ``read`` command into a set of matching email messages, construct json objects representing those messages, and pass them to the Juttle Runtime by calling ``emit()``.
```

Comment:
Somewhere here let's say that most adapter support both `read` and `write` primitives, but the Gmail one only supports `read`... can it do `write` as well, as in, send an email via gmail account? Since we are using it as an example, it would be nice to explain how write is implemented.